cli/run/index.ts

```diff
@@ -118,7 +118,68 @@ async function execute(
 			if (command.stdin !== false) {
 				inputOptions.plugins!.push(stdinPlugin());
 			}
+			if (command.plugin) {
+				const plugins = Array.isArray(command.plugin) ? command.plugin : [command.plugin];
+				for (const plugin of plugins) {
+					if (/[={}]/.test(plugin)) {
+						// -p plugin=value
+						// -p "{transform(c,i){...}}"
+						loadAndRegisterPlugin(inputOptions, plugin)
+					} else {
+						// split out plugins joined by commas
+						// -p node-resolve,commonjs,buble
+						plugin.split(",").forEach((plugin: string) => loadAndRegisterPlugin(inputOptions, plugin));
+					}
+				}
+			}
 			await build(inputOptions, outputOptions, warnings, command.silent);
 		}
 	}
+
+	function loadAndRegisterPlugin(inputOptions: InputOptions, pluginText: string) {
+		let plugin : any = null;
+		let pluginArg : any = undefined;
+		if (pluginText[0] === '{') {
+			// -p "{transform(c,i){...}}"
+			plugin = new Function('return ' + pluginText);
+		} else {
+			const match = pluginText.match(/^([@.\/\\\w|^{}|-]+)(=(.*))?$/);
+			if (match) {
+				// -p plugin
+				// -p plugin=arg
+				pluginText = match[1];
+				pluginArg = new Function('return ' + match[3])();
+			} else {
+				throw new Error(`Invalid --plugin argument format: ${JSON.stringify(pluginText)}`);
+			}
+			if (!/^\.|[@\/\\]/.test(pluginText)) {
+				// Try using plugin prefix variations first if applicable.
+				// Prefix order is significant - left has higher precedence.
+				for (const prefix of ['@rollup/plugin-', 'rollup-plugin-']) {
+					if (!RegExp(prefix).test(pluginText)) {
```

Comment:
I think the only relevant case here is that the pluginText actually starts with one of the prefixes. Also, it makes no sense to try to require `@rollup/plugin-rollup-plugin-node-resolve` or `rollup-plugin-@rollup/plugin-node-resolve`.  So why not change the `if`-statement in line 155 to

```js
if (!['.', '@rollup/plugin-', 'rollup-plugin-'].some(prefix => pluginText.startsWith(prefix)) { //…
```

and get rid of the second `if`.

On further thought, it makes sense IMO to always treat a plugin starting with a `.` as a relative path, so we could treat this case earlier do not branch here if this case applies. Then we could extract `['@rollup/plugin-', 'rollup-plugin-']` as a constant both from the loop and the if-statement.

Reply:
Feel free to take over this PR as long as all the tests pass. The functionality is sufficient for my own projects. The idea was that plugins should use node resolve unless they were relative or absolute paths. The if in question prevents an unnecessary file system lookup in cases were the path is relative or absolute. If you can accomplish that through other simpler means, that's fine too.